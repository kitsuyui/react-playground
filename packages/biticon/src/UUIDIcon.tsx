import { isValidUUID, uuidToBigInt } from "./uuid"
import { BitIcon } from "./BitIcon"
import { applySBox128bitBigintTo128bitBigint } from './bitShuffle'

import type React from "react"

export const UUIDIcon = (props: { uuid: string }): React.JSX.Element => {
  const uuid = props.uuid
  if (!isValidUUID(uuid)) {
    return <span>Invalid UUID</span>
  }
  const bits = uuidToBigInt(uuid)
  const newBits = applySBox128bitBigintTo128bitBigint(bits)
  return <BitIcon bits={newBits} />
}
