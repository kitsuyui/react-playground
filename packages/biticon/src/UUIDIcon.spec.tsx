import { expect, test, describe, } from 'vitest'

import { render } from '@testing-library/react'

import { UUIDIcon } from './UUIDIcon'


describe('UUIDIcon', () => {
  test('renders with valid UUID', () => {
    const uuid = '123e4567-e89b-12d3-a456-426614174000'
    const { asFragment } = render(<UUIDIcon uuid={uuid} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders error message with invalid UUID', () => {
    const uuid = 'invalid-uuid'
    const { asFragment } = render(<UUIDIcon uuid={uuid} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
