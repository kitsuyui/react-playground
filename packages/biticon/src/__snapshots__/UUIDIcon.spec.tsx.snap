// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UUIDIcon > renders error message with invalid UUID 1`] = `
<DocumentFragment>
  <span>
    Invalid UUID
  </span>
</DocumentFragment>
`;

exports[`UUIDIcon > renders with valid UUID 1`] = `
<DocumentFragment>
  <svg
    aria-label="BitIcon: 2121312103121120030302302202021233112022110231100020310130333201"
    height="64"
    role="img"
    viewBox="0 0 64 64"
    width="64"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="0"
      y="0"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="8"
      y="0"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="16"
      y="0"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="24"
      y="0"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="32"
      y="0"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="40"
      y="0"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="48"
      y="0"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="56"
      y="0"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="0"
      y="8"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="8"
      y="8"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="16"
      y="8"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="24"
      y="8"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="32"
      y="8"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="40"
      y="8"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="48"
      y="8"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="56"
      y="8"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="0"
      y="16"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="8"
      y="16"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="16"
      y="16"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="24"
      y="16"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="32"
      y="16"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="40"
      y="16"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="48"
      y="16"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="56"
      y="16"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="0"
      y="24"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="8"
      y="24"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="16"
      y="24"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="24"
      y="24"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="32"
      y="24"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="40"
      y="24"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="48"
      y="24"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="56"
      y="24"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="0"
      y="32"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="8"
      y="32"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="16"
      y="32"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="24"
      y="32"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="32"
      y="32"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="40"
      y="32"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="48"
      y="32"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="56"
      y="32"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="0"
      y="40"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="8"
      y="40"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="16"
      y="40"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="24"
      y="40"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="32"
      y="40"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="40"
      y="40"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="48"
      y="40"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="56"
      y="40"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="0"
      y="48"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="8"
      y="48"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="16"
      y="48"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="24"
      y="48"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="32"
      y="48"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="40"
      y="48"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="48"
      y="48"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="56"
      y="48"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="0"
      y="56"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="8"
      y="56"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="16"
      y="56"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="24"
      y="56"
    />
    <rect
      fill="#fde724"
      height="8"
      width="8"
      x="32"
      y="56"
    />
    <rect
      fill="#35b778"
      height="8"
      width="8"
      x="40"
      y="56"
    />
    <rect
      fill="#440154"
      height="8"
      width="8"
      x="48"
      y="56"
    />
    <rect
      fill="#30678d"
      height="8"
      width="8"
      x="56"
      y="56"
    />
  </svg>
</DocumentFragment>
`;
